src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm

```diff
@@ -310,12 +311,28 @@
     return AsByteSpan(mIntermediateCert);
 }
 
-bool MTROperationalCredentialsDelegate::ToChipEpochTime(uint32_t offset, uint32_t & epoch)
+bool MTROperationalCredentialsDelegate::ToChipNotAfterEpochTime(NSDate * date, uint32_t & epoch)
+{
+    if ([date isEqualToDate:[NSDate distantFuture]]) {
+        epoch = kNullCertTime;
+        return true;
+    }
+
+    return ToChipEpochTime(date, epoch);
+}
+
+bool MTROperationalCredentialsDelegate::ToChipEpochTime(NSDate * date, uint32_t & epoch)
 {
-    NSDate * date = [NSDate dateWithTimeIntervalSinceNow:offset];
     NSCalendar * calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
     NSDateComponents * components = [calendar componentsInTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0] fromDate:date];
 
+    if (!CanCastTo<uint16_t>(components.year)) {
```

Comment:
```suggestion
    if (components.year > kChipEpochMaxYear) {
```
kChipEpochMaxYear (2135) is quite a bit smaller than uint16 range

Reply:
> kChipEpochMaxYear (2135) is quite a bit smaller than uint16 range

It is, but that validation will be done by CalendarToChipEpochTime.  I'd rather not hardcode that dependency here, and just do the one check _this_ code relies on, which is that the year can be losslessly represented in `uint16_t`.

Reply:
I don't think there would be any problems coming out of depending on `kChipEpochMaxYear` here, and your error log text is much more helpful than the silent failure of `CalendarToChipEpochTime`, but I'm fine either way.

Reply:
Ah, the logging I am happy to fix.